src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,83 @@
 import { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (movie: Movie) => void
+};
+
+const emptyMovie = {
+  title: '',
+  description: '',
+  imdbUrl: '',
+  imgUrl: '',
+  imdbId: '',
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const [movie, setMovie] = useState(emptyMovie);
+  // eslint-disable-next-line object-curly-newline
+  const { title, description, imdbId, imdbUrl, imgUrl } = movie;
+  const isDisabled: boolean = !title || !imdbId || !imdbUrl || !imgUrl;
+
+  const setProperty = (property: string, value: string) => {
+    setMovie(prevState => ({
+      ...prevState,
+      [property]: value,
+    }));
+  };
```

Comment:
```suggestion
  const onChange = (event: React.ChangeEvent<HTMLInputElement>) => {
   const { name, value } = event.target;
   
    setMovie(prevState => ({
      ...prevState,
      [name]: value,
    }));
  };
```

use name attribute to change inputField. and pass in TextField only reference to onChange function